src/components/fields/e-signature-field/e-signature-field.tsx

```diff
@@ -0,0 +1,109 @@
+import { Form } from "@lifesg/react-design-system/form";
+import isEmpty from "lodash/isEmpty";
+import { useEffect, useState } from "react";
+import * as Yup from "yup";
+import { AxiosApiClient, FileHelper, generateRandomId } from "../../../utils";
+import { useValidationConfig } from "../../../utils/hooks";
+import { Warning } from "../../shared";
+import { IGenericFieldProps } from "../types";
+import { IESignatureFieldSchema, IESignatureValue } from "./types";
+
+export const ESignatureField = (props: IGenericFieldProps<IESignatureFieldSchema>) => {
+	// =============================================================================
+	// CONST, STATE, REF
+	// =============================================================================
+	const {
+		error,
+		formattedLabel,
+		id,
+		onChange,
+		schema: { label: _label, uiType: _uiType, upload, validation, ...otherSchema },
+		value,
+		warning,
+	} = props;
+	const [stateValue, setStateValue] = useState<IESignatureValue>(value);
+	const [loadingProgress, setLoadingProgress] = useState<number>(null);
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(
+			id,
+			Yup.object()
+				.shape({
+					fileId: Yup.string().required(),
+					dataURL: Yup.string().required(),
+				})
+				.default(undefined),
+			validation
+		);
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [validation]);
+
+	useEffect(() => {
+		setStateValue(value);
+	}, [value]);
+
+	// =============================================================================
+	// EVENT HANDLER
+	// =============================================================================
+	const handleChange = async (signatureDataURL: string) => {
+		const fileId = generateRandomId();
+		if (!isEmpty(upload)) {
+			const response = await uploadFile(fileId, signatureDataURL);
+			onChange({ target: { value: { fileId, dataURL: signatureDataURL, uploadResponse: response } } });
+		} else {
+			onChange({ target: { value: { fileId, dataURL: signatureDataURL } } });
+		}
+	};
+
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const uploadFile = async (fileId: string, signatureDataURL: string) => {
+		setLoadingProgress(0);
+		const formData = new FormData();
```

Comment:
just noting here that drafts `attachments.multipart` endpoint was based on file upload and expects `sessionId` and `slot`, we'll need to update these fields to be optional

Reply:
noted, actually I can add `sessionId` if it's needed but I am looking to do away with `slot`

Reply:
ok to leave these two fields out, they're actually optional on the db level anyway